httpstan/services_stub.py

```diff
@@ -156,5 +156,26 @@ async def call(
         fh.close()
     httpstan.cache.dump_fit(b"".join(compressed_parts), fit_name)
 
+    # if an exception has already occurred, grab relevant info messages, add as context
+    exception = future.exception()
+    if exception:
+        num_context_messages = 4
```

Comment:
Where does this come from?

Reply:
the future tracks the output of the call to the C++ stan sample function. If that function raises a C++ exception then `exception` here will not be `None`.